### ISSUE TITLE
Subtitles renaming doesn't work properly

### ISSUE THREAD

**Issue by martinjgrunwald** [Sonarr/Sonarr#6821]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

When subtitles are imported together with an episode and they are renamed, the renamed subtitles contains parts of the original filename that should be cut

### Expected Behavior

When subtitles are imported together with an episode and they are renamed, the renamed subtitles should have the format <filename>.<language>.<extension>

### Environment

```markdown
- OS: Ubuntu 22.04
- Sonarr: 4.0.4.1668-ls59
- Docker Install: Yes
- Using Reverse Proxy: Traefik
- Browser: Chrome 124
- Database: Sqlite 3.44.2
```

### What branch are you running?

Develop

### Trace Logs?

[sonarr.trace.0.txt](https://github.com/Sonarr/Sonarr/files/15319164/sonarr.trace.0.txt)
[sonarr.trace.1.txt](https://github.com/Sonarr/Sonarr/files/15319165/sonarr.trace.1.txt)
[sonarr.trace.2.txt](https://github.com/Sonarr/Sonarr/files/15319166/sonarr.trace.2.txt)
[sonarr.trace.3.txt](https://github.com/Sonarr/Sonarr/files/15319167/sonarr.trace.3.txt)
[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/15319168/sonarr.trace.txt)

### Anything else?

I used the rename function for Mushoku Tensei S02E18. The original subtitle file name was `[DKB] Mushoku Tensei - Isekai Ittara Honki Dasu - S02E18 [1080p][HEVC x265 10bit][Multi-Subs]_track10.ger`. 
It was renamed to `Mushoku Tensei - Jobless Reincarnation S02E18.Mushoku Tensei - Isekai Ittara Honki Dasu - S02E18 [1080p][HEVC x265 10bit][Multi-Subs]_track10.de`
It should just be `Mushoku Tensei - Jobless Reincarnation S02E18.de`, right?

**Comment by augustuen** [comment]
You've provided too little information for us to tell what the subtitle should've been named. But, the subtitle should match the episode file name, plus tags for language and type.

**Comment by martinjgrunwald** [comment]
I don't really understand what information you need. But once I understand I am willing to give it to you 😄

I think at the moment Sonarr takes some characters from the old file name and appends it to the new one. Is there a setting that influences how the subtitle string is supposed to look like because I couldn't find that

**Comment by martinjgrunwald** [comment]
Yes, it is Anime. This is my Anime Episode Format: `{Series Title} S{season:00}E{episode:00}`
The corresponding video file was renamed to `Mushoku Tensei - Jobless Reincarnation S02E18.mkv` therefore I thought the subtitle should have been renamed to `<video file name>.<language>.<extension>`, i.e. `Mushoku Tensei - Jobless Reincarnation S02E18.de.srt` but that wasn't the case. In the output file name `Mushoku Tensei - Jobless Reincarnation S02E18.Mushoku Tensei - Isekai Ittara Honki Dasu - S02E18 [1080p][HEVC x265 10bit][Multi-Subs]_track10.de.srt` there is way more information in the file name than my format description, right? 
I can give you examples of other shows where the renaming didn't work as expected either. From my understanding this is a bug in the code, which is why I posted it here

**Comment by augustuen** [comment]
Well, I'm not able to replicate the issue with the latest commits. So either the issue has already been fixed, or it's caused by something particular to your setup. Personally, I think it's the latter. So I'll again ask you to join us in one of the support channels (preferably Discord) so we can investigate further. If I'm wrong, it'll be easier to fix the issue after our rounds of troubleshooting.

**Comment by martinjgrunwald** [comment]
Thanks for trying it out yourself, I asked in the Discord channel now

**Comment by markus101** [comment]
Looks like there are a couple different issues here:

1. The "title" isn't parsed from episodes for new imports
2. The title that is parsed just ends up being most of the file name unless it follows a very specific pattern which these files don't

Definitely a bug.

**Comment by WoutVanAelPXL** [comment]
I don't know if this is part of this bug.
But when I renamed my series folder and files it turned all the hi subs into Hindi.

example: 
'moviename.en.hi.srt' turned into 'moviename.hi.srt'

I fixed them back by running 'Remove HI Tags' in Bazarr on those subs:
'moviename.hi.srt' -> 'moviename.en.srt'

Versions:
Version:    4.0.4.1491
Package Version:    4.0.4.1491-ls238 by [linuxserver.io](https://linuxserver.io/)
.NET:     Yes (6.0.13)
Docker: Yes
Database: Sqlite 3.44.2

**Comment by JeWe37** [comment]
Ah, I was worried an oversight like this could happen with that logic... Basically this is caused by cb72e752f9e24e5691292d86ac1f46c8fa35a844 making the possible title separator parsing more lenient, which means false positives become more likely, and the [logic that is supposed to clean those up](https://github.com/Sonarr/Sonarr/blob/66940b283b4a08a37c66298113a6968c746b0e17/src/NzbDrone.Core/MediaFiles/EpisodeImport/Aggregation/Aggregators/AggregateSubtitleInfo.cs#L53) doesn't catch this case, because the subtitle file title isn't a substring of the episode title. 

Really this requires something more complex. Maybe something along the lines: For every possible split between the episode and subtitle title, find the first possible split such that the part before the split is no longer a substring of the episode file title?
